File: src/main/java/com/dh/ondot/schedule/infra/exception/OdsayServiceAreaException.java
package com.dh.ondot.schedule.infra.exception;

import com.dh.ondot.core.exception.BadRequestException;

import static com.dh.ondot.core.exception.ErrorCode.ODSAY_SERVICE_AREA;

public class OdsayServiceAreaException extends BadRequestException {
    public OdsayServiceAreaException(String detail) {
        super(ODSAY_SERVICE_AREA.getMessage().formatted(detail));
    }
    @Override
    public String getErrorCode() {
        return ODSAY_SERVICE_AREA.name();
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/infra/exception/OdsayServiceAreaException.kt
package com.dh.ondot.schedule.infra.exception

import com.dh.ondot.core.exception.BadRequestException
import com.dh.ondot.core.exception.ErrorCode.ODSAY_SERVICE_AREA

class OdsayServiceAreaException(detail: String) :
    BadRequestException(ODSAY_SERVICE_AREA.message.format(detail)) {
    override val errorCode: String get() = ODSAY_SERVICE_AREA.name
}
